processor/memorylimiterprocessor/memorylimiter.go

```diff
@@ -141,6 +145,10 @@ func (ml *memoryLimiter) shutdown(context.Context) error {
 		return errShutdownNotStarted
 	} else if ml.refCounter == 1 {
 		ml.ticker.Stop()
+		ml.closeOnce.Do(func() {
+			close(ml.closed)
+		})
```

Comment:
Don't need this, we guarantee to close only once.

Reply:
without this https://github.com/open-telemetry/opentelemetry-collector/blob/1d0d1ef21baebfcc9ed1d67f0237437ceb9a0049/processor/memorylimiterprocessor/factory_test.go#L28 that test fails

Reply:
I personally think that this test doesn't make sense. I would like @open-telemetry/collector-approvers to say their opinion.

Reply:
I think we can remove the second `Shutdown` call from the test. We don't have guarantees that it's allowed.

Reply:
Oh, it's Start->Shutdown->Start-> causing the failure... This change breaks that workflow even without `ml.closeOnce`. I don't think we have all other components supporting this support.

Reply:
so maybe let's move `closed` and `closeOnce` initialization to `Start` method, the behavior will stay the same

Reply:
> Start->Shutdown->Start

@dmitryax is calling a processor's `Start` after calling `Shutdown` an unsupported action? Would OPAMP ever need to restart a shutdown component?

If that pattern is not allowed I agree that the test is invalid and we can treat the code in `shutdown` as only ever being invoked once.

Reply:
> so maybe let's move closed and closeOnce initialization to Start method, the behavior will stay the same

We can move `closed` channel creation to Start. `closeOnce` will not be needed not needed.

> @dmitryax is calling a processor's Start after calling Shutdown an unsupported action? Would OPAMP ever need to restart a shutdown component?

I'm not sure

> If that pattern is not allowed I agree that the test is invalid and we can treat the code in shutdown as only ever being invoked once.

I'm not saying it's not allowed. We just don't explicitly support it. Likely some components will be failing.

Probably better not not break this pattern here with this change and just move `closed` channel creation to Start.

Reply:
ok, I moved `closed` channel to `Start` and removed `closeOnce`

Reply:
I moved it to `startMonitoring`, so the `closed` channel is created when the goroutine is created. When there are metrics, logs and traces in the pipeline it would be called 3 times and `closed` would be created 3 times too resulting in data race

Reply:
> When there are metrics, logs and traces in the pipeline it would be called 3 times and closed would be created 3 times too resulting in data race

Why? They are synchronized via `refCounterLock`